provisioning/containers/gui-common/websockify/novnc-overrides/app/ui.js

```diff
@@ -1,34 +1,223 @@
+        document.querySelector('#noVNC_connect_dlg .noVNC_logo').innerText = 'Connection failed'
+        connAttempts = 0;
+        oocp();
+    }
 }
+
+
+/**
+ *  STATS VIEW
+ */
+const link = document.createElement('link');
+link.rel = 'stylesheet';
+link.href = 'app/styles/metrics.css';
+document.head.appendChild(link);
+
+class Resources {
+    constructor(cpu, mem, net, ip, timestamp, error) {
+        if (cpu) this.cpu = Number(cpu);
+        if (mem) this.mem = Number(mem);
+        if (net) this.net = Number(net); //ms
+        if (ip) this.ip = ip;
+        if (timestamp) this.timestamp = timestamp;
+        if (error) this.error = error;
+    }
+
+    static from(json) {
+        if (json.cpu) json.cpu = Number(json.cpu) > 100 && 100 || Number(json.cpu);
+        if (json.connInfo) {
+            const infoMap = new Map(Object.entries(json.connInfo));
+            json.ip = infoMap.keys().next().value;
+            json.net = infoMap.get(json.ip); //ms
+        }
+        const resources = Object.assign(new Resources(), json);
+        return resources;
+    }
+}
+
+class ResourcesHistory {
+    constructor(historyLen) {
+        this.cpu = 0;
+        this.mem = 0;
+        this.net = 0;
+        this.currentIndex = 0;
+        this.lastResources = [];
+        this.historyLen = historyLen || 5;
+    }
+
+    /**
+     * Save last resources to compute usages mean. A circular array is used
+     */
+    addResources(resources) {
+        var len = this.lastResources.length;
+        if (this.lastResources.length < this.historyLen) {
+            this.cpu = ((this.cpu * (len)) + resources.cpu) / (len + 1);
+            this.mem = ((this.mem * (len)) + resources.mem) / (len + 1);
+            this.net = ((this.net * (len)) + resources.net) / (len + 1);
+        } else {
+            this.cpu = (((this.cpu * this.historyLen) - this.lastResources[this.currentIndex].cpu + resources.cpu) / this.historyLen) || 0;
+            this.mem = (((this.mem * this.historyLen) - this.lastResources[this.currentIndex].mem + resources.mem) / this.historyLen) || 0;
+            this.net = (((this.net * this.historyLen) - this.lastResources[this.currentIndex].net + resources.net) / this.historyLen) || 0;
+        }
+        this.lastResources[this.currentIndex % this.historyLen] = resources;
+        this.currentIndex = (this.currentIndex + 1) % this.historyLen;
+    }
+}
+
+var resourcesMean = new ResourcesHistory();
+var currentResources = new Resources(null, null, null, null, null, "No fetched resources available");
+
+/**
+ * Register actions needed when page is loaded:
+ *  - WS metrics connection
+ *  - Button drag and drop logic
+ */
+window.addEventListener('load', () => {
+    // WS metrics connection
+    var metricsPath = "ws://" + document.location.host + "/usages?updatePeriod=2";
+    if (window.connUid != undefined) {
+        metricsPath = `${metricsPath}&connUid=${window.connUid}`;
+    }
+    const conn = new WebSocket(metricsPath);
```

Comment:
It would be better to retrieve the metricsPath directly from websockify, as done with websockifyTargetUrl.